File: LeetCode/Medium/2816-double-a-number-represented-as-a-linked-list/2816-double-a-number-represented-as-a-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func doubleIt(head *ListNode) *ListNode {
    if head.Val > 4 { head = &ListNode{0, head} }
    
    curr := head
    
    for curr != nil {
        curr.Val = curr.Val * 2 % 10
        
        if curr.Next != nil && curr.Next.Val > 4 { curr.Val++ }
        curr = curr.Next
    }
    
    return head
}

File: LeetCode/Medium/2816-double-a-number-represented-as-a-linked-list/2816-double-a-number-represented-as-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode doubleIt(ListNode head) {
        if (head.val > 4) head = new ListNode(0, head);
        
        ListNode curr = head;
        
        while (curr != null){
            curr.val = curr.val * 2 % 10;
            
            if (curr.next != null && curr.next.val > 4) ++curr.val;
            curr = curr.next;
        }
        
        return head;
    }
}